sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala

```diff
@@ -589,6 +589,7 @@ case class GpuFileSourceScanExec(
 
 object GpuFileSourceScanExec {
   def tagSupport(meta: SparkPlanMeta[FileSourceScanExec]): Unit = {
+    ShimLoader.getSparkShims.tagHiddenMetadataExpressions(meta)
```

Comment:
We've been moving away from v1 style runtime-loaded shims and instead doing v2 compile-time shims for any new shims.  This could be handled in one of two ways, either adding a v2 shim utility method to check for metadata columns (e.g.: com.nvidia.spark.rapids.shims.v2.DataSourceUtils or whatever) or it can be handled in the meta wrapper for FileSourceScanExec which is already shim-specific (e.g.: see the custom tagging code at https://github.com/NVIDIA/spark-rapids/blob/branch-22.04/sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/Spark30XdbShims.scala#L179-L187), however we'd need to move FileSourceScanExec wrapping to Spark33XShims in that case.

Reply:
I actually started out by copying the `FileSourceScanExec` wrapping into the `Spark33XShims` but it copied a lot of code without modification. I will look into the v2 approach.

Reply:
I removed the call to the shim loader but I'm not sure if my change is exactly what you were looking for. Let me know if I should refactor this to a call to a separate util class.